js/src/forum/components/ComposerBody.js

```diff
@@ -22,45 +22,53 @@ import ItemList from '../../common/utils/ItemList';
  * @abstract
  */
 export default class ComposerBody extends Component {
+  static initProps(props) {
+    super.initProps(props);
+
+    Object.assign(props, props.state.getBody().getAttrs());
+  }
+
   init() {
+    this.state = this.props.state;
+
     /**
      * Whether or not the component is loading.
      *
      * @type {Boolean}
      */
     this.loading = false;
 
-    /**
-     * The content of the text editor.
-     *
-     * @type {Function}
-     */
-    this.content = m.prop(this.props.originalContent);
+    this.state.on('focus', () => {
+      this.focus();
+    });
+    this.state.bodyPreventExit = this.preventExit.bind(this);
+
+    this.state.content(this.props.originalContent || '');
 
     /**
-     * The text editor component instance.
-     *
-     * @type {TextEditor}
+     * @deprecated BC layer, remove in Beta 15.
      */
-    this.editor = new TextEditor({
-      submitLabel: this.props.submitLabel,
-      placeholder: this.props.placeholder,
-      onchange: this.content,
-      onsubmit: this.onsubmit.bind(this),
-      value: this.content(),
-    });
+    this.content = this.state.fields.content;
+    this.editor = this.state;
   }
 
   view() {
-    // If the component is loading, we should disable the text editor.
-    this.editor.props.disabled = this.loading;
-
     return (
       <div className={'ComposerBody ' + (this.props.className || '')}>
         {avatar(this.props.user, { className: 'ComposerBody-avatar' })}
         <div className="ComposerBody-content">
           <ul className="ComposerBody-header">{listItems(this.headerItems().toArray())}</ul>
-          <div className="ComposerBody-editor">{this.editor.render()}</div>
+          <div className="ComposerBody-editor">
+            {TextEditor.component({
+              submitLabel: this.props.submitLabel,
+              placeholder: this.props.placeholder,
+              disabled: this.loading | this.disabled,
```

Comment:
Is that the operator we want? Or should it be `||`?

Reply:
Looks like both work. Preferably `||` because that's the behavior we want.

- `false | false` => `0`
- `false | true` => `1`
- `true | true` => `1`